packages/neptune-extension/src/common/components/input/Input.tsx

```diff
@@ -0,0 +1,64 @@
+// Lib
+import React from 'react';
+import { bemBlock } from 'common/utils/bem';
+
+// Module
+import './Input.less';
+
+interface InputProps {
+  className?: string
+  disabled?: boolean
+  error?: boolean
+  selectOnFocus?: boolean
+  variant?: 'small' | 'big'
+  onFocus?: (ev?: React.FocusEvent<HTMLInputElement>) => void
+}
+
+const block = bemBlock('neptune-input');
+
+const Input: React.FC<InputProps & React.InputHTMLAttributes<HTMLInputElement>> = ({
+  className,
+  disabled,
+  error,
+  selectOnFocus,
+  variant,
+  onFocus,
+  ...props
+}) => {
+  const classNames = block({
+      modifiers: {
+        disabled,
+        error,
+        big: variant === 'big',
+      },
+      extra: className,
+    });
+
+  return (
+    <input
+      {...props}
+      className={classNames}
+      disabled={disabled}
+      onFocus={getOnFocusHandler(onFocus, selectOnFocus)}
+    />
+  );
+};
+
+function selectInputContent(ev: React.FocusEvent<HTMLInputElement>) {
+  ev.target.select();
+}
+
+function getOnFocusHandler(onFocus?: (ev: React.FocusEvent<HTMLInputElement>) => void, selectOnFocus?: boolean) {
```

Comment:
The logic is a bit cluttered here, and the fact that both functions `selectInputContent` and `onFocus` have to be called if provided is hidden.
```suggestion
if (!selectOnFocus) {
  return onFocus;
}
if (!onFocus) {
  return selectInputContent;
}
return () => {} // the combined case.
```